### PR TITLE
add auth to file service

### PR DESCRIPTION
**Pull request by nanoNeel**

### CODE REVIEW

File: src/foam/nanos/blob/FileService.java [hunk 1]
```diff
@@ -61,11 +64,12 @@ protected void download(X x) {
...
       // IDs are unguessable cryptographically strong UUIDs, so no
       // permission check is really necessary.
-      
-// if ( userDAO_.find_(x, file.getOwner()) == null ) {
-//  resp.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-//  return;
-// }
+      User owner = (User) userDAO_.find_(x, file.getOwner());
+      User user = (User) x.get("user");
+      if ( user == null || owner == null || (user.getId()!= owner.getId() && ! auth.check(x,"file.read.*")) ) {
```
Comment (jlhughes) [inline src/foam/nanos/blob/FileService.java]:
```suggestion
      if ( user == null || owner == null || (user.getId() != owner.getId() && ! auth.check(x, "file.read."+file.getd())) ) {
```

Comment (kgrgreer) [inline src/foam/nanos/blob/FileService.java]: Please follow FOAM style guide for spacing.

Comment (kgrgreer) [inline src/foam/nanos/blob/FileService.java]: Better to check file.read.<owner>, so you could have finer control. An admin with the file.read.* permission would still work.

**Review by TW80000 — APPROVED** [review]
LGTM

**Review by jlhughes — APPROVED** [review]

**Pull request merged**